src/experiment.py

```diff
@@ -0,0 +1,22 @@
+from abc import ABC, abstractmethod
+from enum import Enum
+
+
+class ExperimentType(Enum):
+    PROMPT_FLOW = 'Hello Prompt Flow'
+    JUPYTER_NOTEBOOK = 'Hello Jupyter Notebook'
+    PROMPTY = 'Hello Prompty'
+    PYTHON = 'Hello Python'
+
+
+class Experiment(ABC):
+    def __init__(self, name):
+        self.name = name
+
+    @abstractmethod
+    def create_resources(self):
```

Comment:
From an abstraction point of view, maybe our interface can be just `Experiment.create`?

Creating the resources needed, which command to run etc seem to be implementation details that may not need to be exposed?